pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -639,7 +639,7 @@ stdenvNoCC.mkDerivation {
     # no `/usr/include`, there’s essentially no risk to dropping
     # the flag there. See discussion in NixOS/nixpkgs#191152.
     #
-    + optionalString ((cc.isClang or false) && !targetPlatform.isDarwin) ''
+    + optionalString ((cc.isClang or false) && !(cc.isROCm or false) && !targetPlatform.isDarwin) ''
```

Comment:
Could this be e.g. `!targetPlatform.isROCm`? Or is ROCm not actually using the proper cross‐compilation machinery?

LGTM as a temporary fix, anyway.

Reply:
Offload code is treated as an auxiliary arch and doesn't fit into the current cross machinery.

eg:

host, target = x86-64
auxiliary-targets = [gfx908, gfx90a, ...]

auxiliary target code objects are embedded in the same output library/bin in a separate section.

Reply:
I see; sounds complicated :)

This seems fine for now.

Reply:
From what I can tell it isn't using the cross-compilation machinery, but to be honest I know nothing about ROCm (I'm just here because Steam broke). Maybe one of the maintainers can chime in.

Thanks for the review!